Sources/AppcuesKit/Presentation/Debugger/UIDebugger.swift

```diff
@@ -38,7 +38,7 @@ internal class UIDebugger: UIDebugging {
     private let config: Appcues.Config
     private let storage: DataStoring
     private let notificationCenter: NotificationCenter
-    private let analyticsPublisher: AnalyticsPublishing
+    private weak var analyticsPublisher: AnalyticsPublishing?
```

Comment:
curious why this one needs to be weak? Is it because the publisher holds a reference to the debugger in the subscribers?  I _thought_ there was some way to use `weakObjects()` or something like that to have a list of subscribers held weakly (like NotificationCenter does). But I'd have to refresh myself on that one, don't totally remember details.  Just thinking about possible options to avoid mistakes down the line perhaps.

Reply:
>Is it because the publisher holds a reference to the debugger in the subscribers?
Exactly that.

It would be an option for AnalyticsPublisher to weakly hold the decorators and subscribers. A `NSHashTable` would let you do the `weakObjects()` or make the array type a simple wrapper class that holds weak references. This seems like it would be a good next step so I can do a follow up PR with that.